torchmetrics/functional/classification/calibration_error.py

```diff
@@ -0,0 +1,123 @@
+            debias_bins = (acc_bin * (acc_bin - 1) * prop_bin) / (prop_bin * accuracies.size()[0] - 1)
+            ce += torch.sum(torch.nan_to_num(debias_bins))  # replace nans with zeros if nothing appeared in a bin
+        ce = torch.sqrt(ce) if ce > 0 else torch.tensor(0)
+    else:
+        raise ValueError(f"Norm {norm} is not supported. Please select from l1, l2, or max. ")
+    return ce
+
+
+def _ce_update(preds: Tensor, target: Tensor) -> Tuple[Tensor, Tensor]:
+    _, _, mode = _input_format_classification(preds, target)
+
+    if mode == DataType.BINARY:
+        confidences, accuracies = preds, target
+    elif mode == DataType.MULTICLASS:
+        confidences, predictions = preds.max(dim=1)
+        accuracies = predictions.eq(target)
+    elif mode == DataType.MULTIDIM_MULTICLASS:
+        # reshape tensors
+        # for preds, move the class dimension to the final axis and flatten the rest
+        confidences, predictions = torch.transpose(preds, 1, -1).flatten(0, -2).max(dim=1)
+        # for targets, just flatten the target
+        accuracies = predictions.eq(target.flatten())
+    else:
+        raise ValueError(
+            f"Calibration error is not well-defined for data with size {preds.size()} and targets {target.size()}"
+        )
+
+    return confidences, accuracies
+
+
+def calibration_error(preds: Tensor, target: Tensor, n_bins: int = 15, norm: str = "l1"):
+    """
+
+        Computes the top-label calibration error as described in `https://arxiv.org/pdf/1909.10155.pdf`.
+
+        Three different norms are implemented, each corresponding to variations on the calibration error metric.
+
+        L1 norm (Expected Calibration Error)
+
+        .. math::
+            \text{Accuracy} = \frac{1}{N}\sum_i^N \|(p_i - c_i)\|
+
+        Infinity norm (Maximum Calibration Error)
+
+        .. math::
+        \text{Accuracy} =  \max_{i} (p_i - c_i)
+
+        L2 norm (Root Mean Square Calibration Error)
+
+        .. math::
+        \text{Accuracy} = \frac{1}{N}\sum_i^N (p_i - c_i)^2
+
+        Where p_i is the top-1 prediction accuracy in bin i and c_i is the average confidence of predictions in bin i.
+
+        # NOTE: L2-norm debiasing is not yet supported.
+
+
+        Args:
+            preds (Tensor): [description]
+            target (Tensor): [description]
+            n_bins (int, optional): Number of bins to use when computing t. Defaults to 15.
+            norm (str, optional): Norm used to compare empirical and expected probability bins.
+                Defaults to "l1", or Expected Calibration Error.
+            debias (bool, optional): Applies debiasing term, only implemented for l2 norm. Defaults to True.
+    """
+    if norm not in ("l1", "l2", "max"):
+        raise ValueError(f"Norm {norm} is not supported.")
+
+    confidences, accuracies = _ce_update(preds, target)
+
+    bin_boundaries = torch.linspace(0, 1, n_bins + 1)
```

Comment:
I assume that some device casting is needed here
```suggestion
    bin_boundaries = torch.linspace(0, 1, n_bins + 1).to(preds.device)
```